src/Controller/RegistrationController.php

```diff
@@ -37,10 +51,27 @@ public function register(Request $request, UserPasswordEncoderInterface $passwor
             // do anything else you need here, like send an email
 
             return $this->redirectToRoute('home');
+            }
         }
-
+        
         return $this->render('registration/register.html.twig', [
             'registrationForm' => $form->createView(),
+            'captchaMsg' => $message,
         ]);
     }
+    public function captchaverify($recaptcha){
+        $url = "https://www.google.com/recaptcha/api/siteverify";
+        $ch = curl_init();
+        curl_setopt($ch, CURLOPT_URL, $url);
+        curl_setopt($ch, CURLOPT_HEADER, 0);
+        curl_setopt($ch, CURLOPT_RETURNTRANSFER, TRUE); 
+        curl_setopt($ch, CURLOPT_POST, true);
+        curl_setopt($ch, CURLOPT_POSTFIELDS, array(
+            "secret"=>"6LfLOqQUAAAAAKnHelEolkPVmIey1HHKVy6Jhh4X","response"=>$recaptcha));
```

Comment:
This should not be public. Use local environmental variables to get the secret key

Reply:
https://symfony.com/doc/current/configuration/environment_variables.html
Make sure you add the secret code to .env.local, not .env